flask/tests/test_datasets.py

```diff
@@ -210,10 +211,45 @@ def test_delete_dataset_admin(client, test_database, login_as):
     assert client.delete("/api/datasets/1").status_code == 204
     assert client.get("/api/datasets/1").status_code == 404
 
+    # Check length of permission groups
+    assert (len(dataset.groups)) == 0
+
     # Deleting a nonexistent dataset
     assert client.delete("/api/datasets/400").status_code == 404
     assert client.get("/api/datasets/400").status_code == 404
 
+    # Deleting a dataset without an analysis but associated with tissue sample
+    dataset_5 = (
+        models.Dataset.query.filter(models.Dataset.dataset_id == 5)
+        .options(
+            joinedload(models.Dataset.tissue_sample).joinedload(
+                models.TissueSample.datasets
+            ),
+        )
+        .one_or_none()
+    )
+
+    assert client.delete("/api/datasets/5").status_code == 204
+    assert client.get("/api/datasets/5").status_code == 404
+
+    # Check if tissue sample 4 is properly deleted
+    tissue_sample = dataset_5.tissue_sample
+
+    participant = (
+        models.Participant.query.filter(
+            models.Participant.participant_id == tissue_sample.participant_id
+        )
+        .options(joinedload(models.Participant.tissue_samples))
+        .one_or_none()
+    )
+
+    participant_tissue_samples = [
+        s
+        for s in participant.tissue_samples
+        if s.tissue_sample_id == tissue_sample.tissue_sample_id
+    ]
+    assert (len(participant_tissue_samples)) == 0
+
```

Comment:
I suggest making another test function instead to make the intent more clear. For the assertions, you only need to save the tissue_sample_id primary key before the request and then query for that after and assert its nonexistence

Reply:
Ahh I see. Didn't see that we have a GET request for tissue_sample also. I'll clean it up.